logger/LoggerThread.cpp

```diff
@@ -57,6 +64,18 @@ void logStack(const LogItem& item, const char* localTime, bool logStdOut, FILE*
 }
 } // namespace
 
+void LoggerThread::ensureLogFileExists()
+{
+    if (_logFileName.size())
+    {
+        std::ifstream f(_logFileName.c_str());
```

Comment:
I think this is an expensive why to check if a file exist as internally it will open a new file descriptor and initialise an `ifstream` and potential allocate some buffers on these operations. 
I believe a cheaper way to do this is by using `fstat` with file descriptor we already have and you can check how many hard links are for that inode (`st_nlink`). If it is zero, means the file was deleted.

Reply:
Switched to C++17 std::filesystem::exist